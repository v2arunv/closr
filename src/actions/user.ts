import {IUser} from "../models/users";

export interface IUserAction {
    type: string,
    user: IUser,
}

export const setUser = (user: IUser): IUserAction => {
    return <IUserAction> {
        type: 'SET_USER',
        user: user,
    }
};