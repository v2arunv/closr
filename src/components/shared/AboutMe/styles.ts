import {StyleSheet} from "react-native";

const styles = StyleSheet.create({
    row: {
        flexDirection: 'row',
        alignItems: 'center',
        marginTop: 10,
    },
    icon: {
        width: 20,
        height: 20,
    },
    rowText: {
        marginLeft: 10,
    },
    lastRow: {
        marginBottom: 10,
    }

});

export default styles