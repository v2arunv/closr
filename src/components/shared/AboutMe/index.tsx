import {Image, Linking, Text, TouchableOpacity} from "react-native";
import React from "react";
import {IUser} from "@models/users";
import styles from "./styles";
import Section from "@sharedComponents/Section";


interface IProps {
    user: IUser,
}

const openEmail = (email:string) => {
    return () => {
        Linking.canOpenURL(`mailto:${email}`)
            .then((supported) => {
                if (supported) {
                    Linking.openURL(`mailto:${email}`);
                }
            }).catch((err) => {
            console.log('error Linking', err)
        })
    }
};

const openPhone = (phone:string = '*') => {
    return () => {
        Linking.canOpenURL(`tel:${phone}`)
            .then((supported) => {
                if (supported) {
                    Linking.openURL(`tel:${phone}`);
                }
            }).catch((err) => {
            console.log('error Linking', err)
        })
    }
};

const AboutMe = (props: IProps) => {
    const {
        user,
    } = props;
    return(
        <Section>
            <TouchableOpacity
                onPress={openPhone(user.phone)}
                style={styles.row}
            >
                <Image
                    source={require('../../../assets/images/phone-actions-ring.png')}
                    style={styles.icon}
                />
                <Text
                    style={styles.rowText}
                >
                    { user.phone }
                </Text>
            </TouchableOpacity>
            <TouchableOpacity
                style={[styles.row, styles.lastRow]}
                onPress={openEmail(user.email)}
            >
                <Image
                    source={require('../../../assets/images/send-email-2.png')}
                    style={styles.icon}
                />
                <Text
                    style={styles.rowText}
                >
                    { user.email }
                </Text>
            </TouchableOpacity>
        </Section>
    )
};

export default AboutMe;