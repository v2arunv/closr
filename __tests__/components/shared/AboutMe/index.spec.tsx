import React from 'react';
import AboutMe from '@sharedComponents/AboutMe';
import renderer from 'react-test-renderer';
import user from '@mocks/users';

test('renders correctly', () => {
    const tree = renderer.create(<AboutMe user={user}/>).toJSON();
    expect(tree).toMatchSnapshot();
});