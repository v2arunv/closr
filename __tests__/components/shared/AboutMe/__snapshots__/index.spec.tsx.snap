// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
        "marginTop": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
      },
      Object {
        "borderBottomColor": "#ddd",
        "borderBottomWidth": 0,
        "elevation": 1,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 10,
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/phone-actions-ring.png",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
    <Text
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      1-770-736-8031 x56442
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 10,
        "marginTop": 10,
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/send-email-2.png",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
    <Text
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      Sincere@april.biz
    </Text>
  </View>
</View>
`;
